content/news: add GSoC 2022 Work Product

davidchocholaty opened the pull request:
This PR adds the final blog with the work product for the Google Summer of Code 2022. The blog is written for the Pitch Shift effect and Group delay handling project which was implemented by davidchocholaty contributor.

davidchocholaty committed:
pelicanconf: add davidchocholaty gsoc 2022 student

The commit adds David Chocholaty to the developers as a participant
for the GSoC 2022. With that, the avatar for the developer is added
to content/images/avatars.

davidchocholaty committed:
content/news: add GSoC 2022 Work Product

This commit adds the final blog with the work product
for the Google Summer of Code 2022. The blog is written
for the Pitch Shift effect and Group delay handling project
which was implemented by davidchocholaty contributor.

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,651 @@
+title: GSoC 2022 Work Product- Pitch Shift effect and Group delay handling
```
```suggestion
title: GSoC 2022 Work Product - Pitch Shift effect and Group delay handling
```

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,651 @@
+            by increasing pitch is an ascending scale, and a scale ordered
+            by decreasing pitch is a descending scale.</li>
+        </ul> 
+    </li>
```
markdown is simpler and should work just as well.
```suggestion
 * Scale
   * In music theory, a scale is any set of musical notes ordered
    by fundamental frequency or pitch. The scale ordered
    by increasing pitch is an ascending scale, and a scale ordered
    by decreasing pitch is a descending scale.
```

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,651 @@
+to the RubberBand library instead of that the library requires the amount
+of input data.
+
+<hr />
```
```suggestion
---
```

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,651 @@
+status: draft
+tags: TODO(davidchocholaty) add tags
+comments: yes
+
```
I think we should add a disclaimer that this blogpost primarily serves as documentation to google and thus goes into a lot more detail our other blogposts.

davidchocholaty committed:
content/news: fix table borders

This commit fixes the borders of the benchmarking results tables.

davidchocholaty committed:
content/news: extend Things I learned from GSoC

This commit extends the chapter "Things I learned from the GSoC"
with a description of what I learned due to the change of goal
in the project extension.

davidchocholaty committed:
content/news: fix sentence meaning

The commit fixes the meaning of a sentence
in the Acknowledgements chapter.

davidchocholaty committed:
content/news: fix grammar for words separately

This commit fixes the grammar for words, and some of them, are replaced
with better synonyms.

davidchocholaty committed:
content/news: add text and images for Conclusion

The commit adds text and images for the Conclusion chapter.
This chapter summarizes the work done and the contributor's activity
during the GSoC period.

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,673 @@
+Before the effect is implemented, the pitch could be changed using the master
+Pitch Shifter for each deck player separately only. Unfortunately,
+the mentioned Pitch Shifter has no wide range and is not really useful
+for the interesting DJ skills on live sets. In this project, the new Pitch Shift
```
```suggestion
Before the effect is implemented, the pitch of a track could be changed using the decks
rate slider separately only. This imposes significant restrictions in the ways other effects can interact with the sound. In this project, the new Pitch Shift
```

Both of these statements is not really true, there are a lot of creative things DJs have done with the rate adjustment of a turntable. Also it can shift by quite a lot if configured so in the settings.

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,673 @@
+            by fundamental frequency or pitch. The scale ordered
+            by increasing pitch is an ascending scale, and a scale ordered
+            by decreasing pitch is a descending scale.</li>
+        </ul>
```
prefer the simpler Markdown constructs if possible.
```suggestion
* Scale
  * In music theory, a scale is any set of musical notes ordered
    by fundamental frequency or pitch. The scale ordered
    by increasing pitch is an ascending scale, and a scale ordered
    by decreasing pitch is a descending scale.
```

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,673 @@
+    </li>
+    <li>Interval
+        <ul>
+            <li>An interval in music is the distance in pitch
```
"distance in pitch" doesn't make sense to me. What unit is that supposed to be?

davidchocholaty replied:
Replaced with a new version: In music theory, an interval is a difference in pitch between two sounds. In western scales, intervals are most commonly differences between whole tones and semitones.

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,673 @@
+        <ul>
+            <li>A semitone is a distance in pitch between a note
+            and the very next note, higher or lower. It is the smallest interval
+            in western music.</li>
```
```suggestion
            <li>A semitone is a distance in pitch between a note
            and the very next note, higher or lower. It is the smallest interval
            in most western scales.</li>
```

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,673 @@
+            is a data structure that uses a single, fixed-size buffer
+            as if it were connected end-to-end. This structure lends itself
+            easily to buffering data streams.</li>
+        </ul>  
```
I think we should leave citations for snippets ripped from other sources.

davidchocholaty replied:
I'm afraid, that I do not fully understand what you mean. Do you mean to remove the description from here and rather use the link in the text on the citation source?

Swiftb0y replied:
Well, It's common practice when you cite something (just as you cited wikipedia here) you usually provide a reference to the source right where you cited it (or provide at least a footnote). At least thats how I learned it in school in germany, there are probably regional differences though.

Here's an example on how footnotes work in pelican (which we use for building the website): https://github.com/getpelican/pelican/blob/master/pelican/tests/content/article_with_markdown_and_footnote.md

Swiftb0y replied:
So Im just requesting that you add the annotations that link the quote to the source. 
```quote
In computer science, a circular buffer or ring buffer
is a data structure that uses a single, fixed-size buffer
as if it were connected end-to-end. This structure lends itself
easily to buffering data streams.[^1]
```
At the page bottom:
```
[^1]: https://en.wikipedia.org/wiki/Circular_buffer
```

davidchocholaty replied:
Perfect, I just didn't understand, that you have this approach in mind. :+1: Yeah, I know it a lot from technical papers.

davidchocholaty replied:
> At least thats how I learned it in school in germany, there are probably regional differences though.

Just for interest, we use it in the Czech Republic's universities as well.

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,673 @@
+        <ul>
+            <li>An audio buffer holds a single buffer of audio data.
+            The audio buffer size is the amount of time allowed for the computer
+            to process the audio of your sound card or audio interface.</li>
```
```suggestion
            <li>An audio buffer holds a fixed size amount of sampled audio data.
            The audio buffer size determines the time allowed for the computer
            to process the audio data. Thus it also determines Latency.</li>
```

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,673 @@
+        <ul>
+            <li>Latency, from a general point of view, is a time delay
+            between the cause and the effect of some physical change
+            in the system being observed.</li>
```
IMO too general of an explanation. We should either replace it by an audio specific explanation or at least add a concrete example.

davidchocholaty replied:
Replaced with a new version:

In the audio world, “latency” is another word for “delay”.
     The latency (time) of an audio system refers to the time difference
     from the moment a signal is fed into the system, to the moment it appears
     at the output. For example, audio latency is when there’s a noticeable
     delay between the sound being played and the moment it reaches
     the speakers. Depending on the application, such a delay can have various
     effects. Usually, the aim is to achieve the lowest possible latency.

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,673 @@
+            <li>The main branch must build with the packages in the latest
+            Ubuntu LTS which has had at least one point release.</li>
+        </ul>  
+    </li>
```
not important for understanding the article IMO. We should just link to the github wiki page when the term is mentioned in the document body.

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,673 @@
+
+Related files:
+
+- [pitchshifteffect.cpp](https://github.com/mixxxdj/mixxx/blob/main/src/effects/backends/builtin/pitchshifteffect.cpp)
```
we should probably use the permalink instead:
```suggestion
- [pitchshifteffect.cpp](https://github.com/mixxxdj/mixxx/blob/d232119d8fd82e9bdaaf2958e847593156b261f0/src/effects/backends/builtin/pitchshifteffect.cpp)
```

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,673 @@
+#### Demo video
+TODO(davidchocholaty) add demo video
+
+@Video(https://www.youtube.com/watch?v=rt5Ed5GZ1U8)
```
:stuck_out_tongue_winking_eye: :joy: 
```suggestion
@Video(https://www.youtube.com/watch?v=dQw4w9WgXcQ)
```

davidchocholaty replied:
:joy::joy::joy:

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,673 @@
+<hr />
+
+[#4852](https://github.com/mixxxdj/mixxx/pull/4852)
+- RingDelayBuffer: ring buffer for delay handling
```
since these PRs don't belong to the same repo all the time, I'd prepend the repo name.
```suggestion
[mixxx#4852](https://github.com/mixxxdj/mixxx/pull/4852)
- RingDelayBuffer: ring buffer for delay handling
```

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,673 @@
+Added the comments for the Pitch shift effect processing.
+
+<hr />
+
```
I'd also add a mention of the PR to this blog post here since its technically also work done for mixxx imo.

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,673 @@
+audio signals will not overlap based on the latency. With that, the common
+audio processing approach is to delay the original signal by the amount
+of latency to overlap the signals. Based on the Mixxx effect chain API
+(EngineEffectChain), the group delay latency handling was implemented
```
```suggestion
(`EngineEffectChain`), the group delay latency handling was implemented
```
put class names in backquotes so their appear in monospace.

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,673 @@
+Simultaneously, the delay between dynamic Pitch changes is in a smaller range.
+
+*Push and pull model group delay comparison*
+![TODO]({static}/images/news/pitch-shift-effect-graph-1.png)
```
would it be possible to remake these with a color scheme that matches the mixxx website better?
Also it would be nice if these were SVGs if possible.
IMO it would also be better if the y-axis would start at 0 instead of the lowest latency value. That would illustrate why either model is imperfect.

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,673 @@
+for the wet/dry and wet+dry mode. As the last challenge, I would like to mention
+the usage of the std::span from the standard library which is supported
+by C++20. Because the Mixxx organization adheres to the Minimum requirements
+policy for the Ubuntu LTS, the EngineEffectsDelay and RingDelayBuffer
```
this is where I'd link to the github wiki page. Also a slight correction: ubuntu/canonical does not really have any minimum requirements policy as far as I'm aware of. They just include whats stable in their repositories. It's our own policy that we want mixxx to be compatible with the latest ubuntu LTS and thus ensure we stay compatible with the dependencies available in ubuntu.

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,673 @@
+It will allow users to use their own effects they like directly with the support
+of standards such as  LADSPA, DSSI, LV2, VST2, VST3 and so on,
+in the application instead of offering only the built-in effects or poor API
+for the LV2 standard. After consultation with my mentor, we agreed,
+that I will take on this task as a regular Mixxx contributor after the end
+of GSoC.
```
Just to be clear, you're not forced to do anything mixxx related after gsoc. :sweat_smile:   
I was just suggesting that this is probably a very anticipated feature for mixxx and thus worth spending time on.

davidchocholaty replied:
Yeah, I know. :smile:  I would like to continue contributing after GSoC and I took it as an agreement, that I can work on this task due to it being from a similar field as the project.

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,673 @@
+the 27th of 238 contributors for the Mixxx application, with 119 commits
+in total.
+
+*August 2022 mixxxdj/mixxx contributors*
+![TODO]({static}/images/news/august-2022-contributors.png)
+
+*Contributor summary*
+![TODO]({static}/images/news/contributor-summary.png)
```
Can you redo these screenshots with a dark theme to fit the website better? I think you can find that on github > settings > appearance.

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,673 @@
+![TODO]({static}/images/news/contributor-summary.png)
+
+#### Acknowledgements
+First, I would like to many thank my mentor Swiftb0y for his guidance, help,
+reviews, and a lot of new information and lessons he gave me during the summer.
```
Thank you. I'd also like to thank you for spending your valuable time with me on mixxx. While certainly stressful, it has still been fun with you.

You can link my author page if you want.
```suggestion
First, I would like to many thank my mentor [@Swiftb0y]({author}nikolaus-einhauser) for his guidance, help,
reviews, and a lot of new information and lessons he gave me during the summer.
```

davidchocholaty replied:
Thank you very much. Yeah, it was fun for me too. :+1: 

Perfect, I will add it. Is okay that I used your nickname? Or would you like to replace it with your full name as on the developer link? I used the names you and Daniel used on the GSoC page as Mentor's names, so I followed them.

Swiftb0y replied:
I don't really care, my nickname and my real name have become synonymous. Use whatever you like.

Swiftb0y reviewed (changes requested):
Thanks for your patience. I'm sorry I couldn't start reviewing earlier.

davidchocholaty commented:
> Thanks for your patience. I'm sorry I couldn't start reviewing earlier.

No problem, thank you for your reviews. I will fix it with the proposed changes asap. :wink:

davidchocholaty committed:
content/news: remove HTML tags

The commit replaces the HTML tags for lists with the markdown approach
with the star symbol. With that, every "-" symbol for easy lists
is replaced with a star "*" symbol as well. With that, the <hr /> tag
is replaced with "---".

davidchocholaty committed:
content/news: modify the Motivation chapter

The commit modify the description of the deck's slider and why the new
Pitch shift effect should have been implemented.

davidchocholaty committed:
content/news: re-phrase definitions

The commit re-phrases the definitions in the Background chapter.
Specifically, the Semitone and Audio buffer descriptions
were re-phrased.

davidchocholaty committed:
content/news: remake minimum requirements policy

The commit removes the Minimum requirements policy from the background
chapter and the link for Mixxx's description is added to the term
mentioned in the document body.

davidchocholaty committed:
content/news: add repository name

This commit adds the repository name before the pull request hash
to identify for which repository was the pull request created.

davidchocholaty committed:
content/news: put the class names into backquotes

The commit puts the class names into backquotes so they will appear
in monospace.

davidchocholaty committed:
content/news: more specify own policy for Mixxx

The commit specifies more with the word "own", that
the Minimum requirements policy is just own Mixxx's policy
for the Ubuntu LTS.

daschuer commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,632 @@
+and development in general:
+
+* Scale
+  * In music theory, a scale is any set of musical notes ordered
```
This indentation is lost in the rendered html

davidchocholaty replied:
Thank you for the hint. Now, it should be fixed.

daschuer commented:
Sorry for hassle with the deployment. 

This article looks already neat. Just one point: during the first read, I get lost a bit in the technical depths, in the middle part. The "Implementation" catches me again. I have no good suggestion to improve this, but maybe you have an idea?

davidchocholaty commented:
> Sorry for hassle with the deployment.

Absolutely no problem. Thank you very much for the quick fix. :+1:  Just one thing. It looked to me like a bigger issue with Python and I expected it can take a longer time to fix, so, I ignored the Netlify failing checks for a little time and I check the pre-commits locally only. Is it a problem that the Netlify checks for the previous commits have not passed on GitHub? I am not able to re-run the Netlify deployment. The commits should be okay due to the last commit passed without a problem. The previous commits passed locally on pre-commit checks for committing and pushing.

> Just one point: during the first read, I get lost a bit in the technical depths, in the middle part. The "Implementation" catches me again. I have no good suggestion to improve this, but maybe you have an idea?

Maybe I know what you mean. Yeah, it is written in a more technical approach, but IMO it was better to describe what work was done in the PR, than just link without someone knowing, what it is about. You really pointed out the problem, that if somebody without previous knowledge of the implemented structure would read it, it won't make much sense to them. Currently, I can only think that it would not be bad to switch the order of the parts `Implementation` and `Pull requests and issues`. In the `Implementation` chapter, the project is described from a more general point of view and the implemented structures are mentioned too, so the reader is ready to understand the more technical description of the previously described work. I wrote it for those interested in it, but I expect that most people will just skip this part if they would not like to understand it deeply or work on some part in the future.

davidchocholaty committed:
content/news: fix list indentation

The commit fixes the indentation of the list for the Background chapter.
Now, the rendered HTML should be correctly indented as well.

davidchocholaty committed:
content/news: add disclaimer

The commit adds a disclaimer that the blog post primarily serves
as documentation for the Google Summer of Code project.

davidchocholaty committed:
content/news: replace an interval term description

The commit replaces the previous messy description of the term
Interval with a better one.

davidchocholaty committed:
content/news: add a reference to the source

This commit adds the link for the source of the Interval description.

davidchocholaty committed:
content/news: replace latency definition

The commit replaces the definition of Latency with the new
more detailed and in audio approach way. The dependent links are added
to the resources and the old one is removed.

davidchocholaty committed:
content/news: fix links length

The commit fixes links length in Markdown a little bit by entering
the link to the new line.

davidchocholaty committed:
content/news: replace with permalink

The commit replaces the links to files on GitHub
with permalinks instead.

davidchocholaty committed:
content/news: add missing repository name

The commit adds a missing repository name for the link
on the pull request in the plain text part.

davidchocholaty committed:
content/news: add space indent for links

The commit adds a one-space indent for the links that are written
on two rows.

davidchocholaty committed:
content/news: add blog post PR

The commit adds the PR containing the blog post with the "Work Product"
for Google Summer of Code 2022 to the list of PRs.

davidchocholaty committed:
content/news: replace graph images with svg's

The commit replaces graphs with the transparent version in svg format.
The graphs also have the y-axis starting from zero.

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,685 @@
+title: GSoC 2022 Work Product - Pitch Shift effect and Group delay handling
+authors: David Chocholatý
+status: draft
+tags: TODO(davidchocholaty) add tags
```
```suggestion
tags: gsoc, gsoc-2022
```
Any other tags that would fit?

davidchocholaty replied:
If I am thinking about it, just the latency offers, but not sure.

davidchocholaty committed:
content/news: add authors page

This commit adds the author's page for the Mixxx developers
Daniel Schürmann and Nikolaus Einhauser.

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,685 @@
+a critical part of the application engine performance, the tests and benchmarks
+were included in the development.
+
+In this PR, the std::span was newly introduced into the Mixxx app code
```
```suggestion
In this PR, the `std::span` was newly introduced into the Mixxx app code
```

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,685 @@
+- EngineFilterDelay: clamp wrong delay values
+
+Based on the [mixxx#4810](https://github.com/mixxxdj/mixxx/pull/4810) implementation
+it was figured out the bug in the `EngineFilterDelay` structure was. The structure
```
```suggestion
While working on [mixxx#4810](https://github.com/mixxxdj/mixxx/pull/4810) I encountered a bug in the `EngineFilterDelay` structure: The structure
```

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,685 @@
+the Mixxx circular buffer data structure was improved and optimized
+for performance. So, it remains to finish the pull implementation by setting
+the right size of the input ring buffer.  Eventually, implement the input
+ring buffer size depending on the range, that was set. As the last thing,
```
```suggestion
ring buffer size depending on the range that was set. As the last thing,
```

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,685 @@
+and the Pitch knob works in the semitones mode. In musical terminology,
+the pitch is changed based on the semitone chromatic scale. If the toggle
+is off, the Pitch knob works in the continuous mode, as is the default
+for the RubberBand library. As last, the Formant preserving option was added
```
```suggestion
is off, the Pitch knob works in the continuous mode, which is also the default
in the RubberBand library. At last, the Formant preserving option was added
```

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,685 @@
+has a limited range (only 7 semitones up and down in musical terms,
+which means not even an octave), the independent Pitch shift effect offers
+to work in the range of + 2 octaves (+ 24 semitones) and - 2 octaves
+(-24 semitones). The pitch shift effect has the following options:
```
```suggestion
to work in the range of ± 2 octaves (± 24 semitones). 
The pitch shift effect has the following options:
```

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,685 @@
+
+The results of the measurements clearly show, that despite the “pull model”
+implementation for the Pitch shift effect in the effect chain is not optimal,
+it should be preferred before the “push model” implementation.
```
```suggestion
it should be preferred over the “push model” implementation.
```

davidchocholaty committed:
content/news: add tags

The commit adds tags for the blog post. Concretely the gsoc
and gsoc-2022 tags were added.

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,685 @@
+offers the effect to work only with the fixed size audio chunks. Based on that,
+it is not possible to require an amount of input audio data as needed
+for the RubberBand library “pull model” implementation. The SoundTouch library
+for Pitch Shifting was tested too but produced results with the worse
```
```suggestion
for Pitch Shifting was tested too but produced results with worse
```

davidchocholaty committed:
content/news: add Fatih Emre author page

The commit adds a GitHub page for the Fatih Emre GSoC 2022 participant
and Mixxx developer.

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,685 @@
+More widely, as the future work for the Mixxx application, the wider support
+for the LV2 standard for effects can be implemented or better, the Carla
+audio plugin host can be introduced in the Mixxx application.
+It will allow users to use their own effects they like directly with the support
+of standards such as  LADSPA, DSSI, LV2, VST2, VST3 and so on,
```
```suggestion
It will allow users to use their favorite effects enabled via audio plugin
standards such as  LADSPA, DSSI, LV2, VST2, VST3 and so on,
```

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,685 @@
+Mixxx features, the originally proposed extension was replanned and changed.
+The new group delay handling structure was successfully implemented
+and optimized with the implementation of the extended data structure.
+In the GSoC Coding period, the effect was started to optimized and polished
+for minimizing the latency.
```
```suggestion
In the GSoC Coding period, work was started to minimize the effects latency as
well as to polish the effect even more. Unfortunately that work could 
not be finished before the GSoC deadline. 
```

davidchocholaty replied:
Sounds cool, really well-formed. :+1:

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,685 @@
+#### Acknowledgements
+First, I would like to many thank my mentor Swiftb0y for his guidance, help,
+reviews, and a lot of new information and lessons he gave me during the summer.
+I'm just motivated and learned a lot. I would like thank to Mixxx organization
```
```suggestion
I'm just motivated and learned a lot. I would like to thank the Mixxx organization
```

Swiftb0y reviewed (commented):
A couple quick improvements my GF suggested while proofreading.

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,687 @@
+| BM_Copy2WithRampingGain/512         | 116 ns  | 116 ns  | 6011618    |
+| BM_Copy2WithRampingGain/4096        | 1113 ns | 1113 ns | 629003     |
+
+
+#### Demo video
+TODO(davidchocholaty) add demo video
+
+@Video(https://www.youtube.com/watch?v=rt5Ed5GZ1U8)
```
Lets hide this for now until we have the correct video uploaded.
```suggestion

<!---
#### Demo video
TODO(davidchocholaty) add demo video

@Video(https://www.youtube.com/watch?v=rt5Ed5GZ1U8)
-->
```

davidchocholaty committed:
content/news: move citations into annotations

This commit moves the citation resources from the Resources chapter
into the annotations.

davidchocholaty commented:
> A couple quick improvements my GF suggested while proofreading.

Oh, awesome. Tell her I say thank you. :+1:

davidchocholaty committed:
content/news: mark std::span as code

The commit wraps the std::span into backquotes, so it will highlight
on the rendered page as code.

davidchocholaty committed:
content/news: fix grammar and structure of text

The commit fixes the grammar of English text and the structure
of a plain text.

davidchocholaty committed:
content/news: add alternative text for images

The commit finished TODO by adding alternative text for images.

davidchocholaty committed:
content/news: hide the Demo video chapter

This commit hides the Demo video chapter until the correct video
will be uploaded on the Mixxx YouTube channel.

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,714 @@
+* L1 Data 32 KiB (x2)
+* L1 Instruction 32 KiB (x2)
+* L2 Unified 1024 KiB (x2)
+* L3 Unified 36608 KiB (x1)
```
```suggestion
Run on the same system as above.
```

davidchocholaty replied:
Thank you, great idea. :+1:

daschuer commented:
> Is it a problem that the Netlify checks for the previous commits have not passed on GitHub?

That is not a problem. As long as the final commit is working everything is OK. It is unlikely someone will bisect your PR

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,714 @@
+* L1 Instruction 32 KiB (x2)
+* L2 Unified 1024 KiB (x2)
+* L3 Unified 36608 KiB (x1)
+
+Load Average: 1.39, 1.82, 1.75
```
What does this signify? Load on what? What are those numbers?

davidchocholaty replied:
To be honest, I had to do some research about the meaning :smile:. In a nutshell, it describes the computational work of the CPU. More information in the following sources:

https://en.wikipedia.org/wiki/Load_(computing)
https://www.guyrutenberg.com/2008/07/25/understanding-load-average-a-practitioner-guide/

IMO because the pattern is written exactly the same on the Linux system (the `uptime` command), I propose to just keep it as it is for the user who is consecration in it (others will just probably skip this part) or remove it if we agree that it doesn't make sense to publish. Because it is exactly the copy-paste from the Google Benchmark output, I would prefer to keep it as is. IMHO I don't see much value in explaining the reason for the resulting output.

What's your opinion?

Swiftb0y replied:
Interesting, I didn't know google benchmark outputted this info. To be honest, I'd remove them because they just signify load averages over the last 1, 5, 15 minutes. For a benchmark that only runs for 30s, I don't see much value in the info retrieved with the mentioned window sizes.

davidchocholaty replied:
Sounds reasonable, I agree.

davidchocholaty commented:
> > Is it a problem that the Netlify checks for the previous commits have not passed on GitHub?
> 
> That is not a problem. As long as the final commit is working everything is OK. It is unlikely someone will bisect your PR

Okay, perfect. Thank you. :+1:

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,714 @@
+ funny things without extra CPU cycles
+
+More widely, as the future work for the Mixxx application, the wider support
+for the LV2 standard for effects can be implemented or better, the Carla[^2]
```
This footnote seems wrong.

davidchocholaty replied:
Yeah, right. Thank you! I forgot to remove it.

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,714 @@
+     The audio buffer size determines the time allowed for the computer
+     to process the audio data. Thus, it also determines Latency.
+* Latency
+    * In the audio world, “latency” is another word for “delay”[^10].
+     The latency (time) of an audio system refers to the time difference
+     from the moment a signal is fed into the system, to the moment it appears
+     at the output[^11]. For example, audio latency is when there’s a noticeable
+     delay between the sound being played and the moment it reaches
+     the speakers[^10]. Depending on the application, such a delay can have various
+     effects. Usually, the aim is to achieve the lowest possible latency[^11].
+
```
`10` and `11` are here twice, is that on purpose?

davidchocholaty replied:
Yes, it was meant to be. The reason for that is, that the description is a combination of citations from two sources and the whole paragraph made more sense to me if the sentences have combined these ways, so the two citation sources alternate.

Holzhaus commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -0,0 +1,714 @@
+| BM_Copy2WithRampingGain/512         | 116 ns  | 116 ns  | 6011618    |
+| BM_Copy2WithRampingGain/4096        | 1113 ns | 1113 ns | 629003     |
+
+<!---
+#### Demo video
+TODO(davidchocholaty) add demo video
+
+@Video(https://www.youtube.com/watch?v=rt5Ed5GZ1U8)
+-->
```
@Be-ing uploaded the video to our YouTube channel: https://www.youtube.com/watch?v=NTjV7s5Jb_o

davidchocholaty replied:
Awesome. :+1:  Thank you very much for the information and @Be-ing for so fast uploading.

davidchocholaty committed:
content/news: remove wrong footnote usage

The commit removes a forgotten citation, which should be removed.

davidchocholaty committed:
content/news: add a link to the demo video

The commit adds a link to an uploaded video of the Pitch shift effect
demo on the Mixxx YouTube channel.

davidchocholaty committed:
content/news: deduplicate the CPU info

The commit deduplicates the info about the CPU caches for benchmarking
output results. For the second result, it is mentioned, that the system
is the same as previously.

davidchocholaty committed:
content/news: add PR status

The commit adds PR status on GitHub and for WIP PRs, the last commit
that was added in GSoC Coding Period is mentioned and linked.

davidchocholaty committed:
content/news: add what is left to do for #10840

This commit adds information about when the PR will be able to merge.

davidchocholaty committed:
content/news: add a repository to insights names

The commit adds primarily the repository into the description
of the contributor summary insights image.

davidchocholaty committed:
content/news: add Be. into acknowledgements

This commit adds Be. into acknowledgements with thanks for taking care
of publishing the demo video.

davidchocholaty committed:
content/news: improve text with meaning fixes

The commit improves text with some meaning fixes for the Introduction
and Motivation chapters.

davidchocholaty committed:
content/news: improve list items order

The commit improves the order of the items on the list
in the Background chapter.

davidchocholaty committed:
content/news: std::span into backquotes

The commit adds a second mention of the std::span into backquotes
for code-style generation on the page.

davidchocholaty committed:
content/news: improve the Pull requests and issues

This commit improves the Pull requests and issues chapter with
some grammar fixes or updates and adds a definition of the Sample Rate
term into the Background chapter.

davidchocholaty committed:
content/news: improve the Implementation chapter

The commit improves the Implementation chapter. With that, some letter
sizes are adjusted.

davidchocholaty committed:
content/news: improve the rest of the chapters

The commit improves chapters starting the Testing and benchmarking
chapter to the end of the blog post. With that, the links for the GSoC
and SoundTouch library are added.

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -252,49 +290,53 @@ are deduplicated.
 * Formant preserving
 
-The Pitch knob changes the Pitch of the track up or down. For the default
+The Pitch knob changes a pitch of a track up or down. For the default
```
```suggestion
The Pitch knob changes the pitch of a track up or down. For the default
```

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -252,49 +290,53 @@ are deduplicated.
+The Pitch knob changes a pitch of a track up or down. For the default
 middle position, the track pitch is unchanged. The Range knob ensures setting
-the range of the Pitch knob. The Pitch knob based on the range setting can work
+a range of the Pitch knob. The Pitch knob based on the range setting can work
```
```suggestion
the range of the Pitch knob. The Pitch knob based on the range setting can work
```

Swiftb0y commented on content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md:
```diff
@@ -642,7 +676,9 @@ and pull requests. Thank you both for involving me in the Mixxx development
 organization and friend [@Fatih Emre](https://github.com/fatihemreyildiz)
-for his help and synergy on the final blog structure and chapters. Of course,
+for his help and synergy on the final blog structure and chapters. Thank you
+to [@Be.](http://127.0.0.1:8000/news/author/be) for taking care of publishing
```
```suggestion
to [@Be.]({author}be) for taking care of publishing
```

Be-ing replied:
You made the video. Uploading it was trivial. You don't need to credit me for that.

davidchocholaty replied:
Right :smile:

davidchocholaty replied:
Sorry for the possible misunderstanding. The previous deleted ("Right :smile:") answer was for a proposal for change by @Swiftb0y. My web browser just did not update the page. @Be-ing I'm grateful to you that you publish the video so early and made adding it to this PR possible. Of course, I understand if you don't want to be listed, it's your choice. Please for your feedback.

Be-ing replied:
I'd prefer to not be credited here.

davidchocholaty replied:
Okay, no problem. I will fix it.

davidchocholaty committed:
content/news: fix Be. author link

The commit fixes the link for the author.

davidchocholaty committed:
content/news: fix the modified article

The commit fixes the almost modified articles 'a' and 'the' back.

davidchocholaty committed:
content/news: remove load average info

The commit removes the information about the load average of the system
from the benchmark results.

davidchocholaty committed:
content/news: grammar and structure fixes for blog

The commit fixes some grammar and improves a little bit structure
of the blog post.

davidchocholaty commented:
I would like to summarise the status of this PR. From my point of view, the blog post is almost done. I have in mind just last thing. It is about my proposal based on the hint by @daschuer, that it would make maybe sense to switch the order of parts `Pull requests and issues` and `Implementation` (the `Implementation` chapter would move before the `Pull requests and issues` chapter). @Swiftb0y what do you think about it?

Swiftb0y commented:
Sure. Go ahead. IMO its pretty much finished too. I'm just waiting for the final review comments to be resolved. And if you want to switch the section order around, go ahead. :+1:

Swiftb0y commented:
Also you need to rename the blogpost to `XXXX-XX-XX-gsoc-final-blog-david-chocholaty.md` for the date autoupdater update the date on merge. use `git mv` for that instead of regular `mv` so git keeps track of that properly.

Swiftb0y commented:
Rereading the post, I think it would be more appropriate to put `Pull requests and issues` after the `Challenges` section before `Future Work`. What do you think?

davidchocholaty commented:
> Rereading the post, I think it would be more appropriate to put `Pull requests and issues` after the `Challenges` section before `Future Work`. What do you think?

I agree. Sounds cool. :+1:

davidchocholaty committed:
content/news: move Pull requests and issues

This commit moves the Pull requests and issues chapter after
the Challenges chapter (before Future Work).

davidchocholaty committed:
content/news: rename GSoC's final blog post

This commit renames the GSoC final blog post with the XXXX-XX-XX
date pattern.

davidchocholaty commented:
> Also you need to rename the blogpost to `XXXX-XX-XX-gsoc-final-blog-david-chocholaty.md` for the date autoupdater update the date on merge. use `git mv` for that instead of regular `mv` so git keeps track of that properly.

I have used just today's date. Is it right?

Swiftb0y commented:
> > Also you need to rename the blogpost to `XXXX-XX-XX-gsoc-final-blog-david-chocholaty.md` for the date autoupdater update the date on merge. use `git mv` for that instead of regular `mv` so git keeps track of that properly.
> 
> I have used just today's date. Is it right?

Well I literally meant `XXXX-XX-XX` :sweat_smile: See our Readme. https://github.com/mixxxdj/website#adding-new-articles

davidchocholaty commented:
> > > Also you need to rename the blogpost to `XXXX-XX-XX-gsoc-final-blog-david-chocholaty.md` for the date autoupdater update the date on merge. use `git mv` for that instead of regular `mv` so git keeps track of that properly.
> > 
> > 
> > I have used just today's date. Is it right?
> 
> Well I literally meant `XXXX-XX-XX` sweat_smile See our Readme. https://github.com/mixxxdj/website#adding-new-articles

Oh, I understand :smile: Thank you. :+1:

Swiftb0y reviewed (approved):
Thanks. LGTM. I'll merge now so you can submit the post in time.

davidchocholaty commented:
> Thanks. LGTM. I'll merge now so you can submit the post in time.

Perfect, thank you very much for merging and for all the help and proposals for this PR. :+1:

Swiftb0y reviewed (changes requested):
I just realized that its still possible to extend the project timeline. considering there are two open PRs still left, we might want to consider making use of that. I'll hold of from merging until we made a decision.

Swiftb0y reviewed (approved):
Controversy resolved, both @davidchocholaty and me agreed that it doesn't make sense to extend the timeline.

davidchocholaty committed:
content/news: use developer word instead

The commit changes the Mixxx organization administrator designation
to the Mixxx organization developer.
